reporters/reporter-default/src/main/java/org/radargun/reporting/html/ReportDocument.java

```diff
@@ -447,121 +261,131 @@ public boolean accept(Aggregation aggregation) {
-         if (operationThroughput != null) {
-            writeTD(String.format("%.0f&nbsp;reqs/s", operationThroughput.gross), rowStyle);
-            writeTD(String.format("%.0f&nbsp;reqs/s", operationThroughput.net), rowStyle);
-         } else {
-            writeTD("&nbsp;", rowStyle);
-         }
-      }
-      if (presentedStatistics.contains(StatisticType.DATA_THROUGHPUT)) {
-         DataThroughput dataThroughput = operationStats == null ? null : operationStats.getRepresentation(DataThroughput.class);
-         if (dataThroughput != null) {
-            writeTD(String.format("%.0f&nbsp;MB/s - min", dataThroughput.minThroughput / (1024.0 * 1024.0)), rowStyle);
-            writeTD(String.format("%.0f&nbsp;MB/s - max", dataThroughput.maxThroughput / (1024.0 * 1024.0)), rowStyle);
-            writeTD(String.format("%.0f&nbsp;MB/s - mean", dataThroughput.meanThroughput / (1024.0 * 1024.0)), rowStyle);
-            writeTD(String.format("%.0f&nbsp;MB/s - std. dev", dataThroughput.deviation / (1024.0 * 1024.0)), rowStyle);
+      @Property(name = "histogram.buckets", doc = "Number of bars the histogram chart will show. Default is 40.")
+      public int histogramBuckets = 40;
+
+      @Property(name = "histogram.percentile", doc = "Percentage of fastest responses that will be presented in the chart. Default is 99%.")
+      public double histogramPercentile = 99d;
+
+      @Property(name = "histogram.chart.width", doc = "Width of the histogram chart in pixels. Default is 800.")
+      public int histogramWidth = 800;
+
+      @Property(name = "histogram.chart.height", doc = "Height of the histogram chart in pixels. Default is 600.")
+      public int histogramHeight = 600;
+
+      @Property(doc = "Show response time at certain percentiles. Default is 95% and 99%.")
+      public double[] percentiles = new double[]{95d, 99d};
+
+      @Property(doc = "Generate statistics for each node (expandable menu). Default is true.")
+      public boolean generateNodeStats = true;
+
+      @Property(doc = "Generate statistics for each thread (expandable menu). Default is false.")
+      public boolean generateThreadStats = false;
+
+      @Property(doc = "Highlight suspicious results in the report. Default is true.")
+      public boolean highlightSuspects = true;
+   }
+
+   //These methods are used in templates
+   public String img_file(String operation, String suffix, String name) {
+      return String.format("%s%s%s_%s%s_%s", directory, File.separator, testName, operation, suffix, name);
+   }
+
+   public String img(String operation, String suffix, String name) {
+      return String.format("%s_%s%s_%s\"", testName, operation, suffix, name);
+
+   }
+
+   public ArrayList<String> getGeneratedGraphs() {
+      return generatedGraphs;
+   }
+
+   public String histogram(OperationStats operationStats, final String operation, String configurationName, int cluster, int iteration,
```

Comment:
As to histogram & percentileGraph methods: I don't think triggering generation of charts from template is a good idea. This should be invoked from backend logic.